tests/meltano/core/test_utils.py

```diff
@@ -222,3 +223,22 @@ def test_remove_suffix():
     assert remove_suffix("a_string", "ing") == "a_str"
     assert remove_suffix("a_string", "in") == "a_string"
     assert remove_suffix("a_string", "gni") == "a_string"
+
+
+def test_makedirs_decorator(tmp_path):
+    def root(*paths):
+        return tmp_path.joinpath(*paths)
+
+    @makedirs
+    def hierarchy(*ranks):
+        return root(*ranks)
+
+    @makedirs
+    def species(genus_name, species_name):
+        return hierarchy(genus_name, species_name)
+
+    cat = species("felis", "catus")
+    assert cat.exists()
+
+    wolf = species("canis", "lupus", make_dirs=False)
+    assert not wolf.exists()
```

Comment:
Showcase of buggy behavior of `makedirs` before this fix:

```
        wolf = species("canis", "lupus", make_dirs=False)
>       assert not wolf.exists()
E       AssertionError: assert not True
E        +  where True = <bound method Path.exists of PosixPath('/tmp/pytest-of-runner/pytest-0/popen-gw0/test_makedirs_decorator0/canis/lupus')>()
E        +    where <bound method Path.exists of PosixPath('/tmp/pytest-of-runner/pytest-0/popen-gw0/test_makedirs_decorator0/canis/lupus')> = PosixPath('/tmp/pytest-of-runner/pytest-0/popen-gw0/test_makedirs_decorator0/canis/lupus').exists
```

that should've passed.